resources/providers/awslib/eks_provider.go

```diff
@@ -23,32 +23,35 @@ import (
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
 )
 
+type EksCluster types.Cluster
+
 type EksClusterDescriber interface {
-	DescribeCluster(ctx context.Context, clusterName string) (*eks.DescribeClusterResponse, error)
+	DescribeCluster(ctx context.Context, clusterName string) (EksCluster, error)
 }
 
 type EKSProvider struct {
 	client *eks.Client
 }
 
 func NewEksProvider(cfg aws.Config) *EKSProvider {
-	svc := eks.New(cfg)
+	svc := eks.NewFromConfig(cfg)
 	return &EKSProvider{
 		client: svc,
 	}
 }
 
-func (provider EKSProvider) DescribeCluster(ctx context.Context, clusterName string) (*eks.DescribeClusterResponse, error) {
+func (provider EKSProvider) DescribeCluster(ctx context.Context, clusterName string) (EksCluster, error) {
 	input := &eks.DescribeClusterInput{
 		Name: &clusterName,
 	}
-	req := provider.client.DescribeClusterRequest(input)
-	response, err := req.Send(ctx)
+
+	response, err := provider.client.DescribeCluster(ctx, input)
 	if err != nil {
-		return nil, fmt.Errorf("failed to describe cluster %s from eks , error - %w", clusterName, err)
+		return EksCluster{}, fmt.Errorf("failed to describe cluster %s from eks, error - %w", clusterName, err)
 	}
 
-	return response, err
+	return EksCluster(*response.Cluster), err
```

Comment:
I suspect it is a breaking change, and we need to address it in the CSP rules as well/reverting it

For example, https://github.com/elastic/csp-security-policies/blob/main/bundle/compliance/cis_eks/rules/cis_2_1_1/rule.rego

We used to access the cluster field by using `resource.Cluster`

Reply:
yes we discussed it and you said you're ok with it, I can revert though